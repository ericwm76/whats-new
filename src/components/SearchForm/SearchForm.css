.search-form {
  align-items: center;
  display: flex;
  grid-area: search-form;
  justify-content: flex-end;
  justify-self: end;
  width: 50%;
}

.search-input {
  background-color: rgb(215, 241, 255);
  border: 1px solid black;
  border-radius: 5px 0 0 5px;
  font-size: 1em;
  height: 50px;
  padding: 3px;
  width: 60%;
}

.submit-btn {
  background-color: rgb(215, 241, 255);
  border: 1px solid black;
  border-radius: 0 5px 5px 0;
  font-size: 1em;
  height: 50px;
  margin-right: 10px;
  width: 40%;
}

