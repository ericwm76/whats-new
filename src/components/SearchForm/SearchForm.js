import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: ''
    }
  }

  handleChange = (e) => {
    this.setState({ searchInput: e.target.value });
  }

  handleEnterKey = (e) => {
    if (e.keyCode === 13) {
      this.handleSubmit();
    }
  }

  handleSubmit = () => {
    const searchTerm = this.state.searchInput.toLocaleLowerCase();
    this.props.searchNews(searchTerm);
    this.setState({ searchInput: ''});
  }

  render() {
    return (
      <form className="search-form">
        <input 
          className="search-input"
          type="text" 
          name="search" 
          value={this.state.searchInput} 
          placeholder="Search by title or description" 
          onChange={this.handleChange}
          onKeyDown={this.handleEnterKey} 
          />
        <button 
          className="submit-btn"
          type="button"
          onClick={this.handleSubmit}
          >
          Search
        </button>
      </form>
    )
  }
}

export default SearchForm;