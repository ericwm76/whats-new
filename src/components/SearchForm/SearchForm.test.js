import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () =>{
  let mockSearchNews = jest.fn();
  let mockEvent = { target: 
    { value: 'Scooby', keyCode: 13 }
  }
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SearchForm 
        searchNews={mockSearchNews}
      />
    )
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('should call handleChange on change in the input field', () => {
  //   wrapper.find('input').simulate('change');
  // })

  it('should update state when handleChange is called', () => {
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('searchInput')).toEqual('Scooby')
  });

  it('should call searchNews when Submit is clicked', () => {
    wrapper.find('button').simulate('click');
    expect(mockSearchNews).toHaveBeenCalled();
  });

  it('should call searchNews when enter is pressed', () => {
    wrapper.simulate('keyDown');
    expect(mockSearchNews).toHaveBeenCalled();
  });
});