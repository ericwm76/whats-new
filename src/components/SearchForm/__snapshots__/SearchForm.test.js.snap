// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form
  className="search-form"
>
  <input
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search by title or description"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Search
  </button>
</form>
`;
